src/Microsoft.ML.TorchSharp/NasBert/Models/ModelPrediction.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using TorchSharp;
+
+namespace Microsoft.ML.TorchSharp.NasBert.Models
+{
+    internal sealed class ModelForPrediction : NasBertModel
```

Comment:
NERInferenceModel?

Reply:
This isn't for NER. Its for SentenceSimilarity and TextClassification. How about TextModel? TextModelForPrediction? Thoughts?